### PR TITLE
perf: use lowerBound

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: src/headers.js [hunk 1]
```diff
@@ -112,14 +119,13 @@ class Headers {
...
-        break
-      }
+    const i = lowerBound(this[kHeaders], normalizedName)
+    if (this[kHeaders][i] === normalizedName) {
+      this[kHeaders][i + 0] = normalizedName
+      this[kHeaders][i + 1] = normalizedValue
+    } else {
+      this[kHeaders].splice(i, 0, normalizedName, normalizedValue)
     }
-    this[kHeaders].splice(index, 0, normalizedName, normalizedValue)
```
Comment (ronag) [inline src/headers.js:122]: I'm not sure the current implementation here works? What happens if you call `set` on a name that already exists? Should it be inserted or replaced?

Comment (ronag) [reply src/headers.js:122]: Either way I think we are missing a test.

File: src/headers.js [hunk 2]
```diff
@@ -12,10 +12,9 @@ function normalizeAndValidateHeaderName (name) {
 }
 
 function normalizeAndValidateHeaderValue (name, value) {
-  const normalizedHeaderName = normalizeAndValidateHeaderName(name)
   const normalizedHeaderValue = value.replace(/^[\n\t\r\x20]+|[\n\t\r\x20]+$/g, '')
```
Comment (ronag) [inline src/headers.js:15]: Would be nice with a comment of what this regex does. @Ethan-Arrowood

**Comment by Ethan-Arrowood** [comment]
I believe this change has already been made in my v1 pr

**Comment by ronag** [comment]
I don't think what you have in that PR is correct.

**Pull request closed without merging**